### PR TITLE
Reexport signals

### PR DESCRIPTION
**Pull request by blast-hardcheese**
## What do these changes do?

I believe it makes sense to export the underlying exception types to permit consumers to rely on consistent names and packages, so exporting the `Sync`/`Async` `Empty`/`Full` exceptions to consumers.

## Are there changes in behavior for the user?

None, though those who are using LSPs such as pyright will notice warnings such as

```
error: "SyncQueueEmpty" is not exported from module "janus" (reportPrivateImportUsage)
```

will no longer be raised.

## Checklist

- [x] I think the code is well written

### CODE REVIEW

**Comment by blast-hardcheese** [comment]
Also, thank you for the library! I tried to address some of the CI and lint errors as well, but I'm happy to revert those if undesired.

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,8 @@ develop:
 	python setup.py develop
 
 lint flake: checkrst  pyroma bandit mypy
-	flake8 janus tests
+	# F401 is upset about typing.Deque, List, and Set which are used in hinting
```
Comment (Dreamsorcerer) [inline Makefile:5]: The type annotations appear to be ancient. Could we just update them to modern annotations and not disable this?

i.e. `self._queue: Deque[T] = deque()`

Comment (blast-hardcheese) [reply Makefile]: Addressed!

Comment (blast-hardcheese) [reply Makefile]: I also upgraded from `typing.List` to the builtin `list`, as https://peps.python.org/pep-0585/ is supported in 3.9+

**Comment by Dreamsorcerer** [comment]
@asvetlov I saw you released a new upload-coverage yesterday. Do you think you check what's failing here?

**Comment by asvetlov** [comment]
I'll take a look, thanks for the report. 
The previous action version used deprecated and not supported anymore actions for working with GitHub artefacts, thus it was broken.

**Comment by blast-hardcheese** [comment]
@Dreamsorcerer Would you mind kicking the build one more time?

I think I traced down the issue to what seems to be missing sources, which seems to track since we don't `checkout` in the `coverage combine` flow.

**Comment by asvetlov** [comment]
@blast-hardcheese you are correct. I'm working on the fix

**Comment by asvetlov** [comment]
All good now, thanks. 
codecov rate limit is not a blocker.

**Comment by Dreamsorcerer** [comment]
@asvetlov While I have your attention, would you be able to grant access to the domain names on namecheap (same username: Dreamsorcerer)? Last time the domains expired I had to convince namecheap support to let me make a manual payment for renewal. You should be able to add people to a domain by going to "Manage", then "Sharing & Transfer" and add a new manager.

**Pull request merged**